stew/bitops2.nim

```diff
@@ -394,3 +401,147 @@ func rotateRight*(v: SomeUnsignedInt, amount: SomeInteger):
   const mask = bitsof(v) - 1
   let amount = int(amount and mask)
   (v shr amount) or (v shl ( (-amount) and mask))
+
+template mostSignificantBit(T: type): auto =
+  const res = 1 shl (sizeof(T) * 8 - 1)
+  T(res)
+
+template getBit*(x: BitIndexable, bit: Natural): bool =
+  ## reads a bit from `x`, assuming 0 to be the position of the
+  ## least significant bit
+  type T = type(x)
+  (x and T(0b1 shl bit)) != 0
```

Comment:
I can never remember if it still makes a difference, but `bool(x shr bit and 1)` is an alternative which avoids the comparison, afair msvc has better code then or something

Reply:
Addressed in 9ac177d